test/e2e/operator_test.go

```diff
@@ -219,24 +219,49 @@ func TestClusterOperatorStatusRelatedObjects(t *testing.T) {
 			Resource: "namespaces",
 			Name:     "openshift-ingress",
 		},
+		{
+			Group:     iov1.GroupVersion.Group,
+			Resource:  "dnsrecords",
+			Namespace: "openshift-ingress",
+		},
 		{
 			Resource: "namespaces",
 			Name:     "openshift-ingress-canary",
 		},
 	}
 
+	if gatewayAPIEnabled, err := isFeatureGateEnabled(features.FeatureGateGatewayAPI); err != nil {
+		t.Fatalf("Failed to look up %q featuregate: %v", features.FeatureGateGatewayAPI, err)
+	} else if gatewayAPIEnabled {
+		expected = append(expected, configv1.ObjectReference{
+			Group:    "gateway.networking.k8s.io",
+			Resource: "gatewayclasses",
+		})
+		// This test runs before TestGatewayAPI, so we do *not* expect
+		// to see subscriptions, istios, or gateways in relatedObjects.
```

Comment:
If this runs before TestGatewayAPI, what created the gatewayClass?

Reply:
The gatewayapi controller creates the CRD if the featuregate is enabled.  We only need the CRD to add the gatewayclasses resource to `relatedObjects`; we don't need any CR to exist.  (However, if `TestGatewayAPI` does run as part of CI, then having "gatewayclasses" in `relatedObjects` will cause the CI job to gather the gatewayclass that `TestGatewayAPI` created when the CI job runs the must-gather step, after the test step.)